resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala

```diff
@@ -568,6 +568,16 @@ private[spark] object Config extends Logging {
       .checkValue(delay => delay > 0, "delay must be a positive time value")
       .createWithDefaultString("30s")
 
+  val KUBERNETES_MAX_PENDING_PODS =
+    ConfigBuilder("spark.kubernetes.allocation.maxPendingPods")
+      .doc("Maximum number of pending PODs allowed during executor allocation for this " +
+        "application. Those newly requested executors which are unknown by Kubernetes yet are " +
+        "also counted into this limit as they will change into pending PODs by time.")
```

Comment:
Maybe include a note that if there are pending pods for a given resource profile we won't allocate more pods for the rpi since on my first reading of this I assumed this flag changed that behaviour.

Reply:
I was not intend to change the old behavior. So this new limit is global for all the resources profiles within one Spark app because on the Kubernetes point of view it is better to have one global  limit than limiting only the resource profiles separately. In the latter case for example an earlier limit set to N would be multiplied by just changing the app and using an extra resource profile. Moreover if some resource profiles just active in a few stages then the user might need to choose a lower limit but latter when those profiles are not active then the low limit won't be sufficient and the active ones cannot use those free slots.

Reply:
Makes sense, can we still include the note to clarify how this interacts with the existing per-RPI allocations? Since at first read I didn't pick up on that and I imagine other folks might miss that too.